### PR TITLE
Refactoring HubEndPointTests

### PR DESCRIPTION
**Pull request by mikaelm12**
Cleaning up the HubEndPointTests file a little. Moving ~600 lines out of this file.
Getting the Hubs out of this file and adding a a utils class as well.

### CODE REVIEW

File: test/Microsoft.AspNetCore.SignalR.Tests/HubEndpointTestUtils/Utils.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+using System;
+using Microsoft.AspNetCore.SignalR.Internal.Protocol;
+using Xunit;
+
+namespace Microsoft.AspNetCore.SignalR.Tests.HubEndpointTestUtils
+{
+    public class HubEndPointTestUtils
+    {
+        public static Type GetEndPointType(Type hubType)
```
Comment (anurse) [inline test/Microsoft.AspNetCore.SignalR.Tests/HubEndpointTestUtils/Utils.cs]: Is this only used in `IServiceProvider.GetService`? Might be worth a quick refactor to have this actually be a `GetHubEndpoint` helper that takes an `IServiceProvider` and does all the work in one call.

**Comment by anurse** [comment]
Haha! I won the race! (Did I hold back my approval just so I could merge my conflicting change in first? Maaaaaaybe :)). It's a pretty simple conflict though, ping me if you have questions.

File: test/Microsoft.AspNetCore.SignalR.Tests/HubEndpointTestUtils/Utils.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            services.AddOptions()
+                .AddLogging()
+                .AddSignalR();
+
+            addServices?.Invoke(services);
+
+            return services.BuildServiceProvider();
+        }
+
+        public static dynamic GetHubEndpoint(Type hubType)
```
Comment (anurse) [inline test/Microsoft.AspNetCore.SignalR.Tests/HubEndpointTestUtils/Utils.cs]: Ugh... I wish we didn't have to use `dynamic`, but I don't see a smooth way to remove it.

**Comment by mikaelm12** [comment]
Walked right into that one 🤦‍♂️ lol. 
But yeah, it actually makes more sense to merge my change after yours. These conflicts are not bad to resolve.

**Pull request merged**